docker-compose.yml

```diff
@@ -0,0 +1,183 @@
+version: '3.4'
+
+volumes:
+  fcrepo:
+  solr:
+  db:
+  redis:
+  zk1:
+  zkconf:
+  app:
+  assets:
+  cache:
+
+networks:
+  internal:
+
+services:
+  app: &app
+    build:
+      context: .
+      dockerfile: Dockerfile
+      target: DEVELOPMENT_IMAGE
+    image: hyku-api/development_image:latest
+    environment: &web_environment
+      HOST: "0.0.0.0" # Docker cannot forward to a containers localhost, so we need to use this IP instead
+      DATABASE_ADAPTER: postgresql
+      DATABASE_NAME: hyku
+      DATABASE_HOST: db
+      DATABASE_PASSWORD: DatabaseFTW
+      DATABASE_USER: postgres
+      DATABASE_TEST_NAME: hyku_test
+      FEDORA_URL: http://fcrepo:8080/fcrepo/rest
+      LD_LIBRARY_PATH: /opt/fits/tools/mediainfo/linux
+      RAILS_LOG_TO_STDOUT: "true"
+      REDIS_HOST: redis
+      SECRET_KEY_BASE: asdf
+      SETTINGS__ACTIVE_JOB__QUEUE_ADAPTER: sidekiq
+      SETTINGS__BULKRAX__ENABLED: "true"
+      SETTINGS__FITS_PATH: /opt/fits/fits.sh
+      SETTINGS__MULTITENANCY__ADMIN_HOST: hyku.docker
+      SETTINGS__MULTITENANCY__ADMIN_ONLY_TENANT_CREATION: "false"
+      SETTINGS__MULTITENANCY__DEFAULT_HOST: "%{tenant}.hyku.docker"
+      SETTINGS__MULTITENANCY__ROOT_HOST: hyku.docker
+      SETTINGS__MULTITENANCY__ENABLED: "true"
+      SETTINGS__SOLR__URL: http://solr:8983/solr/
+      SETTINGS__ZOOKEEPER__CONNECTION_STR: zookeeper_cluster:2181/configs
+      SOLR_URL: http://solr:8983/solr/
+      PASSENGER_APP_ENV: development
+      GCLOUD_ZONE: 1233465
+      GCLOUD_PROJECT_ID: ABCDEFG
+      PUBSUB_SERVICEACCOUNT_KEY:
+      CHROME_HOSTNAME: chrome # Required for specs setup inside internal_hyku_test rails_helper
+      IN_DOCKER: "true" # Required by internal_test_hyku specs
+      RUBYOPT: "-W:no-deprecated -W:no-experimental" # Turn off Ruby 3.0 deprecation warnings
+      REDIS_NAMESPACE_QUIET: "true" # Turn off Gem deprecations warnings
+    volumes:
+      - app:/home/app/spec/internal_test_hyku/tmp/uploads
+      - assets:/home/app/spec/internal_test_hyku/public/assets
+      - cache:/home/app/spec/internal_test_hyku/tmp/cache
+      - .:/home/app
+    networks:
+      internal:
+
+  web:
```

Comment:
Can any of the services be removed as they're not used by the API?

Reply:
May be looking at the hyrax-orcid docker-compose file will help you  see services you might not need. I looked at it for the docker-compose in hyrax-autopopulation, which you can still check.